### ISSUE TITLE
Can focusOnCloseNode be moved to the accessibility framework?

### ISSUE THREAD

**Issue by samreid** [phetsims/sun#439]
Discovered in #437, there is an a11y option called `focusOnCloseNode` which tracks the currently focused item before opening a dialog, so when the dialog is closed focus can be restored to the prior focused element.

I was wondering if this is something that should be handled more generally by the accessibility engine?  Perhaps it should automatically track a stack of previously focused instances, so that when the stack is popped, it can restore focus without this detail leaking into client instances.  @jessegreenberg has the accessibility team considered this?  I'm not at all sure how it would be implemented.

**Comment by samreid** [comment]
Similar question for this code I saw in Dialog:
```js
          // a11y - store the currently active element before hiding all other accessible content
          // so that the active element isn't blurred
          this.activeElement = this.activeElement || Display.focusedNode;
```

**Comment by jessegreenberg** [comment]
Yes, probably. We reviewed these a while ago, there might be a scenery issue already to improve this, Ill take a look.

**Comment by jessegreenberg** [comment]
Hmm, I can't find anything, but it would be nice to handle this more generally. Ill go ahead and move this to scenery, we will explore this there.
